domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -703,3 +723,113 @@ func (pm *pruningManager) PruneAllBlocksBelow(stagingArea *model.StagingArea, pr
 
 	return nil
 }
+
+func (pm *pruningManager) PruningPointAndItsAnticoneWithMetaData() ([]*externalapi.BlockWithMetaData, error) {
+	onEnd := logger.LogAndMeasureExecutionTime(log, "PruningPointAndItsAnticoneWithMetaData")
+	defer onEnd()
+
+	stagingArea := model.NewStagingArea()
+	pruningPoint, err := pm.pruningStore.PruningPoint(pm.databaseContext, stagingArea)
+	if err != nil {
+		return nil, err
+	}
+
+	pruningPointAnticone, err := pm.dagTraversalManager.AnticoneFromVirtual(stagingArea, pruningPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	blocks := make([]*externalapi.BlockWithMetaData, 0, len(pruningPointAnticone)+1)
+
+	pruningPointWithMetaData, err := pm.blockWithMetaData(stagingArea, pruningPoint)
+	if err != nil {
+		return nil, err
+	}
+
+	blocks = append(blocks, pruningPointWithMetaData)
+	for _, blockHash := range pruningPointAnticone {
+		blockWithMetaData, err := pm.blockWithMetaData(stagingArea, blockHash)
+		if err != nil {
+			return nil, err
+		}
+
+		blocks = append(blocks, blockWithMetaData)
+	}
+
+	return blocks, nil
+}
+
+func (pm *pruningManager) blockWithMetaData(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) (*externalapi.BlockWithMetaData, error) {
+	block, err := pm.blocksStore.Block(pm.databaseContext, stagingArea, blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	daaScore, err := pm.daaBlocksStore.DAAScore(pm.databaseContext, stagingArea, blockHash)
+	if err != nil {
+		return nil, err
+	}
+
+	windowSize := pm.difficultyAdjustmentWindowSize + 1
```

Comment:
It seems like we are always doing `difficultyAdjustmentWindowSize + 1` (If I remember correctly, the reason is because we want difficultyAdjustmentWindowSize intervals, not blocks).

Assuming I'm correct, maybe we want to increment pm.difficultyAdjustmentWindowSize by one, and stop adding 1?